src/ipa_pc/mod.rs

```diff
@@ -69,9 +71,9 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
 
     /// Complete semantic checks on `ck`.
     #[inline]
-    pub fn check_key(ck: &CommitterKey<G>, max_degree: usize) -> bool {
-        let pp = <Self as PolynomialCommitment<G::ScalarField>>::setup(max_degree).unwrap();
-        ck.is_valid() && &pp.hash == &ck.hash
+    pub fn check_key(ck: &CommitterKey<G>, max_degree: usize) -> Result<bool, Error> {
```

Comment:
setup returns a Result for the purpose of being generic in the trait. Here we know that we are in poly commit and the setup cannot fail, so the unwrap was ok. let's restore as it was before